File: infrastructure/db-core/src/main/java/com/nmh/commerce/user/UserRepositoryImpl.java
package com.nmh.commerce.user;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@RequiredArgsConstructor
@Repository
public class UserRepositoryImpl implements UserRepository {
    private final UserJpaRepository userJpaRepository;

    @Override
    public User findById(Long id) {
        return userJpaRepository.findById(id).map(UserEntity::toDomain);
    }
}

File: infrastructure/db-core/src/main/kotlin/com/nmh/commerce/user/UserRepositoryImpl.kt
package com.nmh.commerce.user

import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Repository

@Repository
class UserRepositoryImpl(
    private val userJpaRepository: UserJpaRepository,
) : UserRepository {
    override fun findById(id: Long): User = userJpaRepository.findByIdOrNull(id)?.toDomain()
        ?: throw NoSuchElementException("User not found")

    override fun save(user: User): User = userJpaRepository.save(UserEntity.from(user)).toDomain()
}
